mobile-widgets/qml/main.qml

```diff
@@ -244,13 +244,6 @@ Kirigami.ApplicationWindow {
 				text: qsTr("Dive list")
 				onTriggered: {
 					manager.appendTextToLog("requested dive list with credential status " + prefs.credentialStatus)
-					if (prefs.credentialStatus == CloudStatus.CS_UNKNOWN) {
-						// the user has asked to change credentials - if the credentials before that
-						// were valid, go back to dive list
-						if (prefs.oldStatus == CloudStatus.CS_VERIFIED) {
-							prefs.credentialStatus = prefs.oldStatus
-						}
-					}
```

Comment:
well, this is a fundamental design change. And not one that I necessarily agree with.

Either way, having this in the middle of a PR that says that it removes a certain variable seems... suboptimal

Reply:
The commit message, I believe is pretty clear. Added text in the PR to make it even more clear.

The word "fundamental" seems a bit "over the top". In essence this is caused by the fact, that the old version allowed the user to escape the login, without ever having done a login or choose "no-cloud", something I find is a fundamental design problem. With this PR the user is locked in the login page, until one of 3 things happen:
1) enter verified credentials (divelist is loaded from cloud)
2) select no-cloud mode (divelist is loaded from local)
3) enter new credentials, PIN page shows, enter PIN or cancel (back to enter credentials).

Cancel PIN behaves strongly in the old version. A user enters new credentials, sees the PIN screen, but does not receive the email immediately, so works with other applications (causing the mobile app to close). Having received the email with the PiN, the user starts mobile again, BUT does not see the PIN screen, instead the old credentials are used.

If revert to old credentials is a feature, then at least it should work in all situations (just my opinion).

But of course having read your highly de-motivating opinion of my work, I understand your comment.

Reply:
Yes, "fundamental" is over the top, agreed.

I think your argument here for the behavior change has merit.
As you refactor this (I hope, see my other comments), would you add a shortened version of this to the relevant commit message so that this argument is maintained in the git record?
Thanks

Reply:
tomorrow